core/core.go

```diff
@@ -604,7 +605,7 @@ func (c *Core) WriteBlock(block *types.Block) {
 			return
 		}
 		if order == nodeCtx {
-			parentHeader := c.GetHeader(block.ParentHash(nodeCtx), block.NumberU64(nodeCtx)-1)
+			parentHeader := c.GetHeaderByHash(block.ParentHash(nodeCtx))
```

Comment:
I'm curious why you changed this

Reply:
because that call doesnt work for the genesis block, its probably better to have an exception for genesis but its an optimization, this should work for now